flower/utils/pg_storage.py

```diff
@@ -41,26 +45,49 @@ def event_callback(state, event):
         return
 
     cursor = connection.cursor()
-    try:
-        cursor.execute(_add_event, (
-            datetime.fromtimestamp(event['timestamp']),
-            json.dumps(event)
-        ))
-        connection.commit()
-    except Exception:
-        connection.rollback()
-        raise
-    finally:
-        cursor.close()
+    retries_remaining = REQ_MAX_RETRIES
+    while True:
```

Comment:
It would be more Pythonic to have a *for/else* loop instead of `while True` with "manual" counting:

```python
for _ in range(REQ_MAX_RETRIES):
    ...  # break if successful
else:
   ...  # failed to execute
```